charmhelpers/contrib/hahelpers/apache.py

```diff
@@ -88,3 +105,76 @@ def retrieve_ca_cert(cert_file):
 
 def install_ca_cert(ca_cert):
     host.install_ca_cert(ca_cert, CONFIG_CA_CERT_FILE)
+
+
+def _x509_normalize_input(input_):
```

Comment:
why name this argument `input_`?

Reply:
Primarily because the function is meant to normalize whatever value it receives into a bytestring - predominantly they're supposed to be `unicode/str` and `bytes`, but the last case simply returns a value for any other type that's going to fail downstream from this call.